extensions/azurePublish/src/node/deploy.ts

```diff
@@ -238,6 +238,42 @@ export class BotProjectDeploy {
     }
   }
 
+  private async waitForZipDeploy(token: string, statusUrl: string): Promise<void> {
+    this.logger({
+      status: BotProjectDeployLoggerType.DEPLOY_INFO,
+      message: `Waiting for zip upload processing.`,
+    });
+
+    return new Promise((resolve, reject) => {
+      const timerId = setInterval(async () => {
+        try {
+          const statusResponse = await axios.get(statusUrl, { headers: { Authorization: `Bearer ${token}` } });
```

Comment:
I have an ask out to @tonyanziano on of this token can expire while polling here

Reply:
After looking at the code it looks like ARM tokens _can_ expire during long tasks, and we **do not** refresh these automatically behind the scenes.

This is something we do [for regular access tokens](https://github.com/microsoft/BotFramework-Composer/blob/main/Composer/packages/server/src/services/auth/electronAuthProvider.ts#L124), but it has not been implemented in the [ARM token flow](https://github.com/microsoft/BotFramework-Composer/blob/main/Composer/packages/server/src/services/auth/electronAuthProvider.ts#L63).

I think the reasoning behind this was that logging in to fetch an ARM token usually requires an interactive sign-in and so the UX would not be great for the user if they were suddenly shown a login window without any context when refreshing a token. That being said, I don't think we ever tested this hypothesis out.

We could try using the same token refresh logic with ARM tokens and see what happens, but again, I'm unclear on how the UX would look.

Reply:
During R14 we're revamping the service layer for prov/pub a bit and abstracting the polling for long running operations.  I'll ask Natalie to consider the options for refreshing the token while waiting.

Reply:
Awesome - sounds great